### PR TITLE
Turns trait accessors into defines, fixes some bugs

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
## About The Pull Request

This turns trait accessors into defines, because someone had the bright idea to use them for everything so now they get called millions of times a minute. And proc calls aren't free.

Also provides a cleaner, less side-effect -laden way of accessing traits. If you want to access mind traits, you access mind traits instead of getting them as a side effect when accessing mob traits. 

This also fixes a few bugs such as: quirks duplicating traits when transferring to mobs, presence of roundstart traits preventing the removal of non-roundstart traits, and some traits only giving a null as the trait source.

To be considered: how about standardizing sources as typepaths rather than magic strings? This would allow features like removing all reagent-granted quirks without iterating through all the types of /datum/reagent and collecting a list of IDs. 

## Why It's Good For The Game

It's free performance.

## Changelog
:cl: Naksu
fix: Transferring quirks now properly removes the roundstart trait from the person losing the quirk.
fix: Roundstart traits no longer block the removal of other sources of that trait
code: Trait accessors are now defines rather than functions
/:cl:

### CODE REVIEW

**Comment by XDTM** [comment]
I'd avoid setting the source as a typepath, for anything aside from chemicals there isnt gonna be a clear typepath to use. And traits should not be ever manually manipulated; if you want to remove traits from chems remove the chems causing the traits.

I'm also not hugely happy about how they're being used for quirks in general, but i suppose that if it works it works.

**Comment by vuonojenmustaturska** [comment]
testmerged on basil, doesn't seem to be obviously broken

**Comment by kriskog** [comment]
Nerve grounding was changed to use add/remove trait in pr of mine that just got merged. It needs this treatment.

**Comment by vuonojenmustaturska** [comment]
goddamn tab indentation in the web editor

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -1,3 +1,12 @@
+// trait accessor defines
+#define ADD_TRAIT(target, trait, source) do { var/list/_L; if (!target.status_traits) { target.status_traits = list(); _L = target.status_traits; _L[trait] = list(source); } else { _L = target.status_traits; if (_L[trait]) { _L[trait] |= list(source); } else { _L[trait] = list(source); } } } while (0)
```
Comment (ninjanomnom) [inline code/__DEFINES/traits.dm]: Could you make these multiline so it's actualy legible

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]
This approval has been funded by the local food unsatisfaction committee

**Pull request merged**